src/Simplex/Messaging/Server.hs

```diff
@@ -338,11 +379,14 @@ receive th Client {rcvQ, sndQ, activeAt} = forever $ do
     write q = mapM_ (atomically . writeTBQueue q) . L.nonEmpty
 
 send :: Transport c => THandle c -> Client -> IO ()
-send h@THandle {thVersion = v} Client {sndQ, sessionId, activeAt} = forever $ do
+send h@THandle {thVersion = v} Client {sndQ, sessionId, activeAt} = addLabel . forever $ do
   ts <- atomically $ L.sortWith tOrder <$> readTBQueue sndQ
   void . liftIO . tPut h Nothing $ L.map ((Nothing,) . encodeTransmission v sessionId) ts
   atomically . writeTVar activeAt =<< liftIO getSystemTime
   where
+    addLabel next =
+      myThreadId >>= \tid ->
+        labelThread tid ("smp client $" <> toHex sessionId <> " send") >> next
```

Comment:
```suggestion
    label = "smp client $" <> toHex sessionId <> " send"
```

this sequence seems used quite a few times, so maybe it has to be made a function:

addLabel :: String -> IO ()
addLabel s = myThreadId >>= (`labelThread` s)

Passing next to call to `>>` it seems an anti pattern, you are effectively using "." instead of ">>"